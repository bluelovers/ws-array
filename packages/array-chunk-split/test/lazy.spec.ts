import { arrayChunkSplit, arrayChunkMap, arrayChunkBySize } from '../index';

describe(`arrayChunkSplit`, () =>
{

	test(`3`, () =>
	{

		const arr = [1, 2, 3, 4, 5, 6, 7, 8];

		let actual = arrayChunkSplit(arr, 3);
		let expected = [[1, 2, 3], [4, 5, 6], [7, 8]];

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`2`, () =>
	{

		const arr = [1, 2, 3, 4, 5, 6, 7, 8];

		let actual = arrayChunkSplit(arr, 2);
		let expected = [[1, 2, 3, 4], [5, 6, 7, 8]];

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`4`, () =>
	{

		const arr = [1, 2, 3, 4, 5, 6, 7, 8];

		let actual = arrayChunkSplit(arr, 4);
		let expected = [[1, 2], [3, 4], [5, 6], [7, 8]];

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`1`, () =>
	{

		const arr = [1, 2, 3, 4, 5, 6, 7, 8];

		let actual = arrayChunkSplit(arr, 1);
		let expected = [[1, 2, 3, 4, 5, 6, 7, 8]];

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`5 but will only get 4`, () =>
	{

		const arr = [1, 2, 3, 4, 5, 6, 7, 8];

		let actual = arrayChunkSplit(arr, 5);
		let expected = arrayChunkSplit(arr, 4);

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`8`, () =>
	{

		const arr = [1, 2, 3, 4, 5, 6, 7, 8];

		let actual = arrayChunkSplit(arr, 8);
		let expected = [[1], [2], [3], [4], [5], [6], [7], [8]];

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`empty`, () =>
	{

		const arr = [];

		let actual = arrayChunkSplit(arr, 8);
		let expected = [];

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`throw error`, () =>
	{
		expect(() => arrayChunkSplit([], 0)).toThrowError();
	})

})

describe(`arrayChunkBySize`, () =>
{

	test(`1`, () =>
	{

		const arr = [1, 2, 3, 4, 5, 6, 7, 8];

		let actual = arrayChunkBySize(arr, 1);
		let expected = arrayChunkSplit(arr, arr.length);

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`4`, () =>
	{

		const arr = [1, 2, 3, 4, 5, 6, 7, 8];

		let actual = arrayChunkBySize(arr, 4);
		let expected = arrayChunkSplit(arr, 2);

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`5`, () =>
	{

		const arr = [1, 2, 3, 4, 5, 6, 7, 8];

		let actual = arrayChunkBySize(arr, 5);
		let expected = [[1, 2, 3, 4, 5], [6, 7, 8]];

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`6`, () =>
	{

		const arr = [1, 2, 3, 4, 5, 6, 7, 8];

		let actual = arrayChunkBySize(arr, 6);
		let expected = [[1, 2, 3, 4, 5, 6], [7, 8]];

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`7`, () =>
	{

		const arr = [1, 2, 3, 4, 5, 6, 7, 8];

		let actual = arrayChunkBySize(arr, 7);
		let expected = [[1, 2, 3, 4, 5, 6, 7], [8]];

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`8`, () =>
	{

		const arr = [1, 2, 3, 4, 5, 6, 7, 8];

		let actual = arrayChunkBySize(arr, 8);
		let expected = [[1, 2, 3, 4, 5, 6, 7, 8]];

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`empty`, () =>
	{

		const arr = [];

		let actual = arrayChunkBySize(arr, 8);
		let expected = [];

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`throw error`, () =>
	{
		expect(() => arrayChunkBySize([], 0)).toThrowError();
	})

});
