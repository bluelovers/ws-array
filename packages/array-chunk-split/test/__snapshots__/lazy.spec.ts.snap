// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrayChunkBySize [3, 2, 1] 1`] = `
[
  [
    1,
    2,
    3,
  ],
  [
    4,
    5,
  ],
  [
    6,
  ],
  [
    7,
    8,
  ],
]
`;

exports[`arrayChunkBySize [3, 2, 2] 1`] = `
[
  [
    1,
    2,
    3,
  ],
  [
    4,
    5,
  ],
  [
    6,
    7,
  ],
  [
    8,
  ],
]
`;

exports[`arrayChunkBySize [3, 2, 3] 1`] = `
[
  [
    1,
    2,
    3,
  ],
  [
    4,
    5,
  ],
  [
    6,
    7,
    8,
  ],
]
`;

exports[`arrayChunkBySize [3, 2, 4] 1`] = `
[
  [
    1,
    2,
    3,
  ],
  [
    4,
    5,
  ],
  [
    6,
    7,
    8,
  ],
]
`;

exports[`arrayChunkBySize 1 1`] = `
[
  [
    1,
  ],
  [
    2,
  ],
  [
    3,
  ],
  [
    4,
  ],
  [
    5,
  ],
  [
    6,
  ],
  [
    7,
  ],
  [
    8,
  ],
]
`;

exports[`arrayChunkBySize 4 1`] = `
[
  [
    1,
    2,
    3,
    4,
  ],
  [
    5,
    6,
    7,
    8,
  ],
]
`;

exports[`arrayChunkBySize 5 1`] = `
[
  [
    1,
    2,
    3,
    4,
    5,
  ],
  [
    6,
    7,
    8,
  ],
]
`;

exports[`arrayChunkBySize 6 1`] = `
[
  [
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  [
    7,
    8,
  ],
]
`;

exports[`arrayChunkBySize 7 1`] = `
[
  [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
  ],
  [
    8,
  ],
]
`;

exports[`arrayChunkBySize 8 1`] = `
[
  [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
  ],
]
`;

exports[`arrayChunkBySize empty 1`] = `[]`;

exports[`arrayChunkSplit 1 1`] = `
[
  [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
  ],
]
`;

exports[`arrayChunkSplit 2 1`] = `
[
  [
    1,
    2,
    3,
    4,
  ],
  [
    5,
    6,
    7,
    8,
  ],
]
`;

exports[`arrayChunkSplit 3 1`] = `
[
  [
    1,
    2,
    3,
  ],
  [
    4,
    5,
    6,
  ],
  [
    7,
    8,
  ],
]
`;

exports[`arrayChunkSplit 4 1`] = `
[
  [
    1,
    2,
  ],
  [
    3,
    4,
  ],
  [
    5,
    6,
  ],
  [
    7,
    8,
  ],
]
`;

exports[`arrayChunkSplit 5 but will only get 4 1`] = `
[
  [
    1,
    2,
  ],
  [
    3,
    4,
  ],
  [
    5,
    6,
  ],
  [
    7,
    8,
  ],
]
`;

exports[`arrayChunkSplit 8 1`] = `
[
  [
    1,
  ],
  [
    2,
  ],
  [
    3,
  ],
  [
    4,
  ],
  [
    5,
  ],
  [
    6,
  ],
  [
    7,
  ],
  [
    8,
  ],
]
`;

exports[`arrayChunkSplit empty 1`] = `[]`;
